### PR TITLE
allow floats as weights

### PR DESCRIPTION
**Pull request by georgewhewell**
Don't know if float weights were deliberately prohibited, trivial test case passes though.

### CODE REVIEW

File: lib/edge.ex [hunk 1]
```diff
@@ -48,7 +48,7 @@ defmodule Graph.Edge do
     label = Keyword.get(opts, :label)
     weight = Keyword.get(opts, :weight, 1)
     case {label, weight} do
-      {_, w} = meta when is_integer(w) ->
+      {_, w} = meta when is_integer(w) or is_float(w) ->
```
Comment (bitwalker) [inline lib/edge.ex]: Can you replace all `is_integer(w) or is_float(w)` in this PR with `is_number(w)`?

**Comment by bitwalker** [comment]
Sorry for the delay, just one change required and I'll get this merged!

**Comment by bitwalker** [comment]
Actually, it's fine, I can get it taken care of :)

**Comment by georgewhewell** [comment]
@bitwalker thanks!

**Pull request merged**